execution/executors/shim/process.go

```diff
@@ -35,8 +35,14 @@ func newProcess(ctx context.Context, o newProcessOpts) (*process, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
```

Comment:
in this case, however err is only overriden with new values though, so only defer block is enough